homeassistant/components/nanogrid_air/api.py

```diff
@@ -0,0 +1,64 @@
+"""Fetch API for Nanogrid Air."""
+
+import logging
+import socket
+
+import aiohttp
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def get_ip():
+    """Fetch IP address from hostname."""
+    try:
+        ip = socket.gethostbyname("ctek-ng-air.local")
+        _LOGGER.debug("Resolved IP: %s", ip)
+        return ip
+    except socket.gaierror as e:
+        _LOGGER.error("Failed to resolve hostname: %s", e)
+        return
+
+
+async def fetch_mac():
+    """Fetch mac address from status API."""
+    ip = await get_ip()
+
+    url_status = f"http://{ip}/status/"
+
+    async with aiohttp.ClientSession() as session:
+        try:
+            async with session.get(url_status) as api_status_response:
+                api_status_response.raise_for_status()
+                mac = await api_status_response.json()
+                mac_address = mac["deviceInfo"]["mac"]
+                return mac_address
+        except aiohttp.ClientError as exc:
+            _LOGGER.error("HTTP client error occurred in fetch_mac: %s", exc)
+            return {}
+        except aiohttp.HttpProcessingError as exc:
+            _LOGGER.error(
+                "HTTP request error occurred: %s - %s", exc.status, exc.message
+            )
+            return {}
+
+
+async def fetch_meter_data():
+    """Fetch data from the API and return as a dictionary."""
+    ip = await get_ip()
```

Comment:
Each data update results in the need for a hostname lookup. Seems we should be able to assume the resolved IP to remain and only perform a new lookup if for nay reason are unable to reach the device as expected?

Reply:
Fixed